charts/cryostat/templates/deployment.yaml

```diff
@@ -194,6 +194,31 @@ spec:
           securityContext:
             {{- toYaml .Values.storage.securityContext | nindent 12 }}
         {{- if not .Values.minimal }}
+        - name: {{ printf "%s-%s" .Chart.Name "oauth2proxy" }}
+          securityContext:
+            {{- toYaml .Values.oauth2Proxy.securityContext | nindent 12 }}
+          image: "{{ .Values.oauth2Proxy.image.repository }}:{{ .Values.oauth2Proxy.image.tag }}"
+          args:
+            - "--alpha-config=/etc/oauth2_proxy/alpha_config/alpha_config.yaml"
+          imagePullPolicy: {{ .Values.oauth2Proxy.image.pullPolicy }}
+          env:
+            - name: OAUTH2_PROXY_REDIRECT_URL
+              value: "http://localhost:8080/oauth2/callback"
+            - name: OAUTH2_PROXY_COOKIE_SECRET
+              value: {{ include "cryostat.cookieSecret" . }}
+            - name: OAUTH2_PROXY_EMAIL_DOMAINS
+              value: "*"
```

Comment:
I think this also needs environment variables to tell it to use the `htpasswd` and what permissions to assign to users it finds there, like this:

https://github.com/cryostatio/cryostat3/blob/a85f4900250969578a93526fd4bbe61e3fbe4ef2/smoketest/compose/auth_proxy.yml#L41